flink-streaming-java/src/test/java/org/apache/flink/streaming/api/operators/async/AsyncWaitOperatorTest.java

```diff
@@ -771,9 +773,11 @@ public void testClosingWithBlockedEmitter() throws Exception {
 		MockEnvironment environment = createMockEnvironment();
 
 		StreamTask<?, ?> containingTask = mock(StreamTask.class);
+		TaskMailboxImpl mailbox = new TaskMailboxImpl();
 		when(containingTask.getEnvironment()).thenReturn(environment);
 		when(containingTask.getCheckpointLock()).thenReturn(lock);
 		when(containingTask.getProcessingTimeService()).thenReturn(new TestProcessingTimeService());
+		when(containingTask.getTaskMailboxExecutor(any())).thenReturn(new MailboxExecutorImpl(mailbox));
```

Comment:
Yes, but afaik, it would require quite a bit of refactoring, which I'd do in another PR.

Reply:
Dropping mockito here should be easy:
```
+import org.apache.flink.streaming.util.MockStreamTask;
+import org.apache.flink.streaming.util.MockStreamTaskBuilder;
 import org.apache.flink.streaming.util.OneInputStreamOperatorTestHarness;
 import org.apache.flink.streaming.util.TestHarnessUtil;
 import org.apache.flink.util.ExceptionUtils;
@@ -770,10 +772,10 @@ public class AsyncWaitOperatorTest extends TestLogger {
 
                MockEnvironment environment = createMockEnvironment();
 
-               StreamTask<?, ?> containingTask = mock(StreamTask.class);
-               when(containingTask.getEnvironment()).thenReturn(environment);
-               when(containingTask.getCheckpointLock()).thenReturn(lock);
-               when(containingTask.getProcessingTimeService()).thenReturn(new TestProcessingTimeService());
+               MockStreamTask containingTask = new MockStreamTaskBuilder(environment)
+                       .setCheckpointLock(lock)
+                       .setProcessingTimeService(new TestProcessingTimeService())
+                       .build();
 
                StreamConfig streamConfig = new MockStreamConfig();
                streamConfig.setTypeSerializerIn1(IntSerializer.INSTANCE);
```
and adding a new setter and accessor for `getMailboxExecutor(int ignored) -> mailbox`

If it's indeed that simple I would prefer it as `hotfix` commit in this PR. Preferably as a base commit, before your changes, so that there is no intermittent code that will be removed in a commit after. But if you do not like the rebase hassle, I would be also ok in a `hotifx` commit on top of your change.

Reply:
Done